#!/usr/bin/env python
# Copyright (c) 2016 The UUV Simulator Authors.
# All rights reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

PACKAGE = "uuv_auv_teleop"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("axis_thruster", int_t, 0, "Joystick axis used for thruster", 1, 0, 8)
gen.add("axis_roll", int_t, 0, "Joystick axis used for roll", 0, 0, 8)
gen.add("axis_pitch", int_t, 0, "Joystick axis used for pitch", 4, 0, 8)
gen.add("axis_yaw", int_t, 0, "Joystick axis used for yaw", 3, 0, 8)

gen.add("n_fins", int_t, 0, "Number of fins", 4, 0, 8)

gen.add("scale_thruster", double_t, 0, "Scale used for thruster", 500, 0, 1000.)
gen.add("scale_roll", str_t, 0, "Scale for roll", "1, 1, 1, 1")
gen.add("scale_pitch", str_t, 0, "Scale used for pitch", "1, 1, -1, -1")
gen.add("scale_yaw", str_t, 0, "Scale used for yaw", "-1, 1, 1, -1")

gen.add("thruster_topic",    str_t,    0, "Thruster topic",  "thruster/input")

gen.add("fin_topic_prefix",    str_t,    0, "Prefix of fin topic",  "fins/")
gen.add("fin_topic_suffix",    str_t,    0, "Suffix of fin topic",  "/input")

exit(gen.generate(PACKAGE, "uuv_auv_teleop", "JoyAnglesThrust"))
